--- crosstool-ng-1.5.2.org/Makefile.in	2016-03-07 13:13:33.734405836 +0100
+++ crosstool-ng-1.5.2/Makefile.in	2016-03-07 13:14:57.116400376 +0100
@@ -24,9 +24,14 @@
 # Note: dual test, because if -R and -r are given on the command line
 # (who knows?), MAKEFLAGS contains 'Rr' instead of '-Rr', while adding
 # '-Rr' to MAKEFLAGS adds it literaly ( and does not add 'Rr' )
+# Further: quad test because the flags 'rR' and '-rR' can be reordered.
 ifeq ($(filter Rr,$(MAKEFLAGS)),)
 ifeq ($(filter -Rr,$(MAKEFLAGS)),)
+ifeq ($(filter rR,$(MAKEFLAGS)),)
+ifeq ($(filter -rR,$(MAKEFLAGS)),)
 CT_MAKEFLAGS += -Rr
+endif # No -rR
+endif # No rR
 endif # No -Rr
 endif # No Rr
 
