diff -Naur ACE_wrappers/ace/os_include/os_sched.h ACE_wrappers-patched/ace/os_include/os_sched.h
--- ACE_wrappers/ace/os_include/os_sched.h	2009-04-17 16:04:36.000000000 +0200
+++ ACE_wrappers-patched/ace/os_include/os_sched.h	2018-06-06 15:12:54.441440812 +0200
@@ -36,7 +36,7 @@
 {
 #endif /* __cplusplus */
 
-#if !defined (__cpu_set_t_defined) || !defined (ACE_HAS_CPU_SET_T)
+#if !defined (ACE_HAS_CPU_SET_T)
 #if defined (ACE_HAS_CPUSET_T)
    typedef cpuset_t cpu_set_t;
 #else
